docs/arrow-functions.md

```diff
@@ -70,51 +70,51 @@ setTimeout(person.growOld,1000);
 growOld();
 ```
-If you are going to call it yourself, i.e.
+Eğer fonksiyonu kendiniz çağıracaksanız, örneğin;
 ```ts
 person.growOld();
 ```
-then `this` is going to be the correct calling context (in this example `person`).
+İşte o zaman `this` doğru bir içeriği çağırmış olur (`person` örneğindeki gibi).
 
-#### Tip: Arrow Function Danger
+#### İpucu: Tehlikeli Ok Fonksiyonu Kullanımı
 
-In fact if you want `this` *to be the calling context* you should *not use the arrow function*. This is the case with callbacks used by libraries like jquery, underscore, mocha and others. If the documentation mentions functions on `this` then you should probably just use a `function` instead of a fat arrow. Similarly if you plan to use `arguments` don't use an arrow function.
+Aslında `this`'i *ilgili içeriği çağırmak için* istiyorsanız, *ok fonksiyonu kullanmamalısınız*. jquery, underscore, mocha gibi geri çağırımlar (callback) kullanan kütüphaneler bu duruma örnektir. Eğer dökümanında fonksiyonlarda `this` kullanımından bahsediyorsa, o zaman kalın ok yerine muhtemelen `function` kullanmanız gerekiyor. Aynı şekilde eğer `arguments` kullanmayı düşünüyorsanız da, kalın ok fonksiyonu kullanmamalısınız.
 
-#### Tip: Arrow functions with libraries that use `this`
-Many libraries do this e.g. `jQuery` iterables (one example http://api.jquery.com/jquery.each/) will use `this` to pass you the object that it is currently iterating over. In this case if you want to access the library passed `this` as well as the surrounding context just use a temp variable like `_self` like you would in the absence of arrow functions.
+#### İpucu: Ok fonksiyonları ve `this` kullanan kütüphaneler
+Birçok kütüphane bunu yapmaktadır. Örneğin, `jQuery` yineleyicileri (iterable) (bir örneği http://api.jquery.com/jquery.each/) o an yinelenmekte olan nesneye `this`'i iletecektir. Bu durumda, `this`'i ve çevreleyen içeriğini ileten bir kütüphaneye erişmek istiyorsanız, benzeri bir geçici değişken kullanın: 
 
 ```ts
 let _self = this;
 something.each(function() {
-    console.log(_self); // the lexically scoped value
-    console.log(this); // the library passed value
+    console.log(_self); // sözcük anlamı olarak kapsamın değeri
+    console.log(this); // kütüphanenin ilettiği değer
 });
 ```
 
-#### Tip: Arrow functions and inheritance
+#### İpucu: Ok fonksiyonları ve kalıtım
 
-If you have an instance method as an arrow function then it goes on `this`. Since there is only one `this` such functions cannot participate in a call to `super` (`super` only works on prototype members). You can easily get around it by creating a copy of the method before overriding it in the child.
+Ok fonksiyonu ile yazılıp `this` ile devam eden, bir sınıfa ait bir metodunuz var. Sadece tek bir `this` olduğundan,  bu tip fonksiyonlar `super`'i (`super` sadece prototip üyelerinde çalışır) çağıramazlar. Alt sınıf metodun üzerinden geçmeden önce, metodun kopyasını yaratarak bunun üstesinden gelebilirsiniz.
```

Comment:
Cümlenin başına Diyelim ki deyip devam etsem kulağa nasıl geliyor? Cümlecikleri birbirine yapıştırınca genelde okunması/ anlaşılması zorlaşıyor. Önerdiğim hali şu:

"Diyelim ki Ok fonksiyonu ile yazılıp `this` ile devam eden, bir sınıfa ait bir metodunuz var. Sadece tek bir `this` olduğundan, ..."